### PR TITLE
Adding support for custom network namespace in the gateway operator

### PR DESCRIPTION
**Pull request by alacuku**
# Description

Supporting the creation and configuration of a new network namespace for the gateway operator. The idea is to move the gateway pod in host network. In order to not put the iptables configuration in the node's namespace a new one is created. In the custom network namespace we put all the liqo configuration such as vpn tunnel and iptables rulespecs.

Fixes #(issue)

# How Has This Been Tested?

Unit tests have been written for all the new functions.

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -222,7 +222,9 @@ github.com/containerd/typeurl v0.0.0-20190911142611-5eb25027c9fd/go.mod h1:GeKYz
 github.com/containerd/typeurl v1.0.1/go.mod h1:TB1hUtrpaiO88KEK56ijojHS1+NeF0izUACaJW2mdXg=
```
Comment (palexster) [inline go.sum:222]: Can you perform a go mod tidy?

File: pkg/liqonet/netns/netns.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	}
+	var createVethPair = func(hostNS ns.NetNS) error {
+		_, _, err := ip.SetupVethWithName(originVethName, dstVethName, linkMTU, dstNetns)
+		if err != nil {
+			klog.Errorf("an error occurred while creating veth pair between host and gateway namespace: %v", err)
+			return err
+		}
+		return nil
+	}
+	retryiable := func(err error) bool {
```
Comment (palexster) [inline pkg/liqonet/netns/netns.go:71]: Why do you need a retriable here?

**Review by palexster — APPROVED** [review]

File: pkg/liqonet/netns/netns.go (hunk 2 above)
Comment (alacuku) [reply pkg/liqonet/netns/netns.go:76]: The veth pair has one one device on the host network and the other one on the custom namespace. Some times if we delete the custom namespace it wold take some time for the kernel to remove the veth device on the host network, and a conflict could arise while reinstalling the veth device in the host network. I also added a comment.

File: pkg/liqonet/routing/common.go [hunk 3]
```diff
@@ -2,8 +2,12 @@ package routing
 
 import (
 	"errors"
+	"io/ioutil"
 	"net"
 	"reflect"
+	"strings"
+
+	"k8s.io/client-go/util/retry"
```
Comment (aleoli) [inline pkg/liqonet/routing/common.go:10]: this import can be joined with the last import block

**Review by aleoli — APPROVED** [review]

**Pull request merged**